lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
So yeah, I've already asked about putting this point in some sort of errata/known issues for the release notes fwiw, most probably as a leading ticket as well people can vote/watch. Mainly, so its less surprising/documented somewhere.